src/test/groovy/org/prebid/server/functional/tests/SetUidSpec.groovy

```diff
@@ -329,4 +388,142 @@ class SetUidSpec extends BaseSpec {
         where:
         bidderName << [UNKNOWN, WILDCARD, GENERIC_CAMEL_CASE, ALIAS, ALIAS_CAMEL_CASE]
     }
+
+    def "PBS should remove all cookies when incoming request have specified optout flad"() {
+        given: "Setuid request"
+        def request = SetuidRequest.defaultSetuidRequest
+        def genericUidsCookie = UidsCookie.getDefaultUidsCookie(GENERIC).tap {
+            optout = false
+        }
+
+        when: "PBS processes setuid request"
+        singleCookiesPbsService.sendSetUidRequest(request, genericUidsCookie)
+
+        then: "Request should fail with error"
+        def exception = thrown(PrebidServerException)
+        assert exception.statusCode == 400
+        assert exception.responseBody == 'Sync is not allowed for this uids'
+    }
+
+    def "PBS should merge cookies when incoming request have multiple uids cookies"() {
+        given: "Setuid request"
+        def request = SetuidRequest.defaultSetuidRequest.tap {
+            uid = UUID.randomUUID().toString()
+        }
+        def genericUidsCookie = UidsCookie.getDefaultUidsCookie(GENERIC)
+        def rubiconUidsCookie = UidsCookie.getDefaultUidsCookie(RUBICON)
+
+        when: "PBS processes setuid request"
+        def response = multipleCookiesPbsService.sendSetUidRequest(request, [genericUidsCookie, rubiconUidsCookie])
+
+        then: "Response should contain requested uids"
+        assert response.uidsCookie.tempUIDs[GENERIC]
+        assert response.uidsCookie.tempUIDs[RUBICON]
+
+        and: "Headers uids cookies should contain same cookie as response"
+        def setUidsHeaders = getSetUidsHeaders(response)
+        def uidsCookie = extractHeaderTempUIDs(setUidsHeaders.first)
+        assert setUidsHeaders.size() == 1
+        assert uidsCookie.tempUIDs[GENERIC]
+        assert uidsCookie.tempUIDs[RUBICON]
+    }
+
+    def "PBS should sout multiple uids cookies by priority and  expiration timestamp"() {
```

Comment:
typo: send*

`PBS should send multiple uids cookies by priority and expiration timestamp`